app/models/manageiq/providers/vmware/infra_manager/orchestration_template.rb

```diff
@@ -60,6 +62,15 @@ def deployment_spec(opts)
     deploy_options
   end
 
+  def workflow_helper
+    @workflow_helper ||= MiqProvisionOrchWorkflow.new({:src_vm_id => [id]}, User.current_user, :skip_dialog_load => true, :initial_pass => true)
+  end
+
+  def target_name_valid?(name)
+    # TODO: need a way to tell the ovf template is for VM template (where target is a VM) or vApp template (where target is a resource pool with VMs)
+    Vm.find_by(:name => name, :ems_id => ext_management_system.id).blank?
```

Comment:
We should be checking for vm names in the same target folder, there is no problem with two VMs with the same name in different folders

Reply:
What does the query look like? I don't see any `ems_folder` information in a `Vm` object.

Reply:
Well assuming you have a `target_folder` (not sure of the exact name just an example) object something like this gets the names of the vms in a folder:
```
>> folder = vc.ems_folders.find_by(:name => "agrare")
>> folder.children.select { |c| c.kind_of?(VmOrTemplate) }.map(&:name)
=> ["ag_content_library_test", "ag_miq_0", "manageiq-vsphere-devel", "ag_test_template", "ag_blank_template", "ag_prov_test_7", "manageiq-jansa-1"]
```